CONTRIBUTING.md

```diff
@@ -34,64 +24,93 @@ The services are deployed as the same docker `arkhn/river` image, but are run wi
+|---------------------------------|-----------------------------------------------------------------------------------------|
+| `docker-compose.yml`            | The minimal functional configuration. `fhir-api` and `pyrog-api` URLs must be provided. |
+| `docker-compose.monitoring.yml` | Optional monitoring services. Configuration is store in the root `monitoring` directory |
+| `docker-compose.test.yml`       | Optional services for testing (e.g. `mimic`)                                            |
 
-You must provide `fhir-api` and `pyrog-api` URLs (which means you have them deployed somewhere)
-
-#### `docker-compose.monitoring.yml`
-
-Contains optional monitoring services.
+### Notes on environment
 
-The services config files are in the root `monitoring` directory.
+- Environment variables can be stored in a `.env` file in the root directory.
+- The `.env` file is shared and loaded by:
+  - `docker-compose`,
+  - Django's `manage.py` script,
+  - Vscode's integrated terminal and launch commands.
 
-#### `docker-compose.test.yml`
+## Backend development
 
-Optional test services.
+### Local prerequisites
 
-### Notes on environment
+```bash
+# 1. Create a virtual env
+python3.8 -m venv --prompt "river" .venv
 
-- Environment variables can be stored in a `.env` file in the root directory.
-- The `.env` file is loaded by:
-  - Django's `manage.py` script, in development,
-  - Vscode's integrated terminal and launch commands,
-  - Docker-compose's cli.
+# 2. Activate the virtual env
+source .venv/bin/activate
 
-## Local development
+# 3. Install dev requirements
+pip install -r requirements/dev.txt
+```
 
-### Local prerequisites
+### Developing with the development server
 
-1. Create a virtual env
+This concerns the API. Not the ETL services (which are not web applications).
 
-   python3.8 -m venv --prompt "river" .venv
+First, you'll need to provide configuration by creating a dotenv file (`.env`).
 
-2. Activate the virtual env
+```bash
+# 1. Copy the dotenv file. The template should be enough to get you started.
+cp .template.env .env
 
-   source .venv/bin/activate
+# 2. Bring the db up
+docker-compose up -d db
 
-3. Install dev requirements
+# 3. Migrate
+python django/manage.py migrate
 
-   pip install -r requirements/dev.txt
+# 4. Run the development server
+python django/manage.py runserver
+```
 
 ### Code quality
 
 Code quality is enforced with `pre-commit` hooks: `black`, `isort`, `flake8`
 
-1. Install the hooks
-
-   pre-commit install
-
+```bash
+# Install the hooks
+pre-commit install
+```
 ### Tests
 
-1.  Run tests in dedicated virtual env.
-
-        tox
+```bash
+# Run tests in dedicated virtual env
+tox
+```
 
-## OpenAPI schema generation
+### OpenAPI schema generation
 
 Generation only concerns the `pyrog` application at this point. To generate the schema (no virtual env required):
 
       tox -e openapi
 
 This produces a `pyrog-schema.yml` file in the root project directory.
```

Comment:
Do you think we should generate both this file and the rtk-query generated file with this command line? Or add some kind of versioning (like Django does with its migration files) telling the frontend team the generated schema is out of date ?

Reply:
1. lets keep front and back toolings as decoupled as possible.
2. the client can be generated (either manually or automatically by a hook or by the CI in a new commit) with the PR that contains the back implementation.
3. the client is published as package
4. the front does not use the source directly, but adds a dependency

We should not have 2. without 3 and 4, or a back PR could break the front. If we dont have 3 and 4, then Id say that the front is in charge of maintaining the client.

I think we can simply rely on git for the versioning.